spec/checkPullRequestLabelsSpec.js

```diff
@@ -0,0 +1,269 @@
+      id: 1,
+      cert: 'test',
+      githubToken: 'test',
+    });
+
+    app = robot.load(oppiaBot);
+    spyOn(app, 'auth').and.resolveTo(github);
+  });
+
+  describe('when pull request gets labeled', () => {
+    describe('assigns based on changelog label', () => {
+      const label = {
+        id: 638839900,
+        node_id: 'MDU6TGFiZWw2Mzg4Mzk5MDA=',
+        url: 'https://api.github.com/repos/oppia/oppia/labels/PR:%20released',
+        name: 'PR CHANGELOG: Server Errors -- @kevintab95',
+        color: '00FF00',
+      };
+      beforeEach(async () => {
+        // Set the payload action and label which will simulate adding
+        // the changelog label.
+        payload.payload.action = 'labeled';
+        payload.payload.label = label;
+        spyOn(checkPullRequestLabelModule, 'checkAssignee').and.callThrough();
+        await robot.receive(payload);
+      });
+
+      it('should call pull request label module', () => {
+        expect(checkPullRequestLabelModule.checkAssignee).toHaveBeenCalled();
+      });
+
+      it('should assign the project owner', () => {
+        expect(github.issues.addAssignees).toHaveBeenCalled();
+        const params = {
+          repo: payload.payload.repository.name,
+          owner: payload.payload.repository.owner.login,
+          number: payload.payload.number,
+          assignees: ['kevintab95'],
+        };
+        expect(github.issues.addAssignees).toHaveBeenCalledWith(params);
+      });
+      it('should ping the project owner', () => {
+        expect(github.issues.createComment).toHaveBeenCalled();
+        const params = {
+          repo: payload.payload.repository.name,
+          owner: payload.payload.repository.owner.login,
+          number: payload.payload.number,
+          body:
+            'Assigning @kevintab95 for the first-pass review' +
+            ' of this pull request. Thanks!',
+        };
+        expect(github.issues.createComment).toHaveBeenCalledWith(params);
+      });
+    });
+
+    it('does not assign project owner if he is the pr author', async () => {
+      const label = {
+        id: 638839900,
+        node_id: 'MDU6TGFiZWw2Mzg4Mzk5MDA=',
+        url: 'https://api.github.com/repos/oppia/oppia/labels/PR:%20released',
+        name: 'PR CHANGELOG: Server Errors -- @kevintab95',
+        color: '00FF00',
+      };
+
+      // Set the payload action and label which will simulate adding
+      // the changelog label.
+      payload.payload.action = 'labeled';
+      payload.payload.label = label;
+      // Set project owner to be pr author.
+      payload.payload.pull_request.user.login = 'kevintab95';
+      spyOn(checkPullRequestLabelModule, 'checkAssignee').and.callThrough();
+      await robot.receive(payload);
+
+      expect(checkPullRequestLabelModule.checkAssignee).toHaveBeenCalled();
+      expect(github.issues.addAssignees).not.toHaveBeenCalled();
+    });
+
+    it('does not assign if another label got added', async () => {
```

Comment:
another label got added -> changelog label is not added

Reply:
Done.